pkgs/games/hmcl/default.nix

```diff
@@ -0,0 +1,95 @@
+{ lib
+, stdenv
+, fetchurl
+, makeWrapper
+, makeDesktopItem
+, wrapGAppsHook
+, copyDesktopItems
+, imagemagick
+, jre
+, xorg
+, libGL
+, glfw
+, openal
+, libglvnd
+, alsa-lib
+, wayland
+, libpulseaudio
+}:
+
+let
+  version = "3.5.5";
+  icon = fetchurl {
+    url = "https://github.com/huanghongxun/HMCL/raw/release-${version}/HMCLauncher/HMCL/HMCL.ico";
+    hash = "z7pc97nxz89gf42fm3pbd0mib2qdf67g";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "hmcl";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://github.com/huanghongxun/HMCL/releases/download/release-${version}/HMCL-${version}.jar";
+    hash = "sha256-bXZF38pd8I8cReuDNrZzDj1hp1Crk+P26JNiikUCg4g=";
+  };
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [
+    makeWrapper
+    wrapGAppsHook
+    copyDesktopItems
+    imagemagick
+  ];
+
+  dontWrapGApps = true;
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "HMCL";
+      exec = "hmcl";
+      icon = "hmcl";
+      comment = meta.description;
+      desktopName = "HMCL";
+      categories = [ "Game" ];
+    })
+  ];
+
+  installPhase = let
+    libpath = with xorg; lib.makeLibraryPath ([
```

Comment:
We try to avoid using `with`. Even this is a valid `nix` statement, we try to avoid it.

There are several compelling reasons to avoid using a `with` statement.

Firstly, employing such a statement imports all the functions from `xorg` into your current scope. This can potentially cloud your code's clarity and lead to confusion.

Secondly, the `with` statement could interfere with your Language Server Protocol (LSP). Given that the entire `xorg` scope is imported, your LSP might struggle to pinpoint the sources of various functions, resulting in a poor development experience.

There are additional considerations regarding this topic that fall outside my expertise, but I encourage you to post this question on discourse for a more comprehensive discussion.

You can find a very good article about that on [nix.dev](https://nix.dev/recipes/best-practices#with-scopes).

There is also this RFC: https://github.com/r-burns/nixos-rfcs/blob/rfc-inherit-as-list/rfcs/0110-inherit-as-list.md

Reply:
OK, I used it